shared/src/main/scala/squants/thermal/Temperature.scala

```diff
@@ -141,21 +141,15 @@ object Temperature extends Dimension[Temperature] with BaseDimension {
   def apply[A](n: A, scale: TemperatureScale)(implicit num: Numeric[A]) = new Temperature(num.toDouble(n), scale)
 
   def apply(s: String): Try[Temperature] = {
-    val regex = "([-+]?[0-9]*\\.?[0-9]+(?:[eE][-+]?[0-9]+)?)[ °]*(f|F|c|C|k|K|r|R)".r
+    val regex = "([-+]?[0-9]*\\.?[0-9]+(?:[eE][-+]?[0-9]+)?)*[ ]*°? *(f|F|c|C|k|K|r|R)".r
```

Comment:
For consistency, I suggest one of the following:
- enclose space in square brackets both times, i.e. `[ ]*°?[ ]*`
- don't space in square brackets, i.e. ` *°? *`
- take it to eleven and allow all the goofy varieties of horizontal whitespace including tab and unicode, i.e. `[\\p{Zs}\\t]*°?[\\p{Zs}\\t]*`

(See http://www.fileformat.info/info/unicode/category/Zs/list.htm for the meaning of `\p{Zs}`)